File: ScaleCalculator/ScaleCalculator/FlatSurface.cpp
#include "FlatSurface.h"

FlatSurface::FlatSurface(const std::vector<Point>& corners, LineSegment measure):Region(corners)
{
	this->measure = measure;
}

Point FlatSurface::buildEndpoint_approximate(Point start, LineSegment direction)
{
	double length = direction.length * (measure.GetPixelLength() / measure.length);
	return direction.BuildCroppedSegmentFromPoint(start, length).end;
}

void FlatSurface::Deserialize(const json& segment)
{
	this->measure.Deserialize(segment["measure"]);
}

File: MetadataGenerator/CPPclasses/FlatSurface.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CPPclasses
{
    public class FlatSurface : Region
    {
        public LineSegment measure { get; set; }
        public FlatSurface(List<Point> corners, LineSegment measure) : base(corners)
        {
            this.measure = measure;
        }

        public FlatSurface() { }

        // Унаследовано через Region
        public override Point buildEndpoint_approximate(Point start, LineSegment direction)
        {
            double length = direction.length * (measure.GetPixelLength() / measure.length);
            return direction.BuildCroppedSegmentFromPoint(start, length).end;
        }
    };
}
